benchmark/Benchmark.h

```diff
@@ -100,14 +157,24 @@ class BenchmarkRecords {
      *  Most of the time a lambda, that calls the actual function to benchmark
      *  with the needed parameters.
      */
+    template<typename FUNCTION_TYPE>
     void addToExistingGroup(const std::string& groupDescriptor,
-        const std::string descriptor,
-        const std::function<void()>& functionToMeasure);
+        const std::string& descriptor,
+        const FUNCTION_TYPE& functionToMeasure) {
+      // Does the group exist?
+      auto groupEntry = recordGroups_.find(groupDescriptor);
+      AD_CHECK(groupEntry != recordGroups_.end());
```

Comment:
Maybe this should throw a "real" exception that says, that the group with the descriptor was not found and that you previously have to call `addGroup`.
(similar for the `addToExistingTable` function, and maybe also for the `addGroup` and `addTable` (for the checks whether a group is already there.

Reply:
I already replaced this part with `.at()`, which throws a runtime exception, if the key isn't already in use. Is this alright, or should I change it back, and add a custom runtime exception?